src/prefect/utilities/storage.py

```diff
@@ -82,22 +82,24 @@ def extract_flow_from_file(
         raise ValueError("Provide either `file_path` or `file_contents`.")
 
     # Load objects from file into dict
-    exec_vals = {}  # type: ignore
+    # if a file_path has been provided, provide __file__ as a global variable
+    # so it resolves correctly during extraction
+    exec_vals = {"__file__": file_path} if file_path else {}  # type: ignore
```

Comment:
Unrelated to your PR: Can we audit if `file_contents` is used and perhaps just simplify by dropping it? I feel like composing something with `extract_flow_from_source(code: str, variables: dict = None)` would make more sense than rolling these into a single method as has been done here.

Reply:
`file_contents` is used quite frequently for other storage, I'm happy to refactor if you'd like but I'd want to do so in another PR

Reply:
Alas. This actually makes sense for remote file based storage so I don't see refactoring it happening anytime soon.